src/core/utility/file/filePath/FilePath.cpp

```diff
@@ -426,28 +408,24 @@ bool FilePath::hasExtension(const std::vector<std::wstring>& extensions) const {
 }
 
 FilePath& FilePath::operator=(const FilePath& other) {
-  m_path = std::make_unique<boost::filesystem::path>(other.getPath());
+  m_path = std::make_unique<fs::path>(other.getPath());
   m_exists = other.m_exists;
-  m_checkedExists = other.m_checkedExists;
   m_isDirectory = other.m_isDirectory;
-  m_checkedIsDirectory = other.m_checkedIsDirectory;
   m_canonicalized = other.m_canonicalized;
   return *this;
 }
 
 FilePath& FilePath::operator=(FilePath&& other) noexcept {
   m_path = std::move(other.m_path);
   m_exists = other.m_exists;
-  m_checkedExists = other.m_checkedExists;
   m_isDirectory = other.m_isDirectory;
-  m_checkedIsDirectory = other.m_checkedIsDirectory;
   m_canonicalized = other.m_canonicalized;
   return *this;
 }
 
 bool FilePath::operator==(const FilePath& other) const {
   if(exists() && other.exists()) {
-    return boost::filesystem::equivalent(getPath(), other.getPath());
+    return fs::equivalent(getPath(), other.getPath());
```

Comment:
Please use std::errorCode not to throw

Reply:
There is a problem:
Philosophy of std::filesystem is to throw exception while there is an issue or pass an error code variable to hold the error and user is free in his next step.
Question if we modified all the code to use error code technique?  --> Will we proceed or terminate? I think we will proceed and let caller decide.
Modifying all code in FilePath & FileSytem to use error code technique, or just use noexcept keyword (silent death).

Reply:
Need to take an action.